### DISCUSSION TITLE
Autocomplete: Clearing selection using SelectedValueChanged

### DISCUSSION THREAD

**Discussion by Hugo-Matias** [Megabit/Blazorise#4587] [Q&A] [answered]
I have a shared _List\<string\>_ of selected values that is populated by Autocomplete in default selection mode and another source as well.

I'm achieving this by using the event callback **SelectedValueChanged** instead of **@bind-SelectedValue**. My goal now is to clear the input field after a value is selected from the dropdown and passed into the OnSelectionChanged handler.

[On the documentation](https://blazorise.com/docs/extensions/autocomplete) I noticed there is a **Clear** method but I'm not sure how to use it. Should I create a reference to the Autocomplete component and call it on the list manipulation method? How do I do that?

**Comment by David-Moreira** [comment, marked as answer]
Hello @Hugo-Matias 
Yes, that is how you call methods on the component, add 
`@ref="myAutocompleteRef"` to the component.
and 
```
@code{
private Autocomplete<MyTItem,MyTValue> myAutocompleteRef;

private Task Clear(){
return myAutocompleteRef.Clear();
}
}
```

Something along those lines, you might have to fix syntax.

**Reply by Hugo-Matias** [reply]
Ah I see, thanks for clarifying David. Still learning the basics and that might be useful in the future as well. Cheers!
